File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = {}  # val -> index

        for i, n in enumerate(nums):
            diff = target - n
            if diff in prevMap: #checks for key
                return [prevMap[diff], i] #finds uses the diff as the key to get the index value
            prevMap[n] = i #updates the dict with elements and its respective index

File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int mid = nums.length / 2;
        int l = 0;
        int r = nums.length - 1;

        while (l <= r) {
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] < target) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }

            mid = l + (r - l) / 2;
        }

        return -1;
    }
}
